e2e/tests/nightly/admin/user-profile-admin.spec.ts

```diff
@@ -0,0 +1,469 @@
+
+    // Verify Access module names displayed correctly
+    const moduleNames = await accountAccessTable.getRowValues(1);
+    expect(moduleNames.sort()).toEqual(accessModules.sort());
+    expect(moduleNames.length).toEqual(accessModules.length);
+
+    // check badges
+    // $x('//*[@id="Registered-user"]')
+    // $x('//*[@id="Controlled-user"]')
+  });
+
+  test('Verify CANCEL and SAVE buttons work correctly', async () => {
+    const userProfileAdminPage = new UserProfileAdminPage(adminTab);
+    await userProfileAdminPage.waitForLoad();
+
+    // Verify Account access toggle text label update after toggle
+    const accountAccessSwitch = userProfileAdminPage.getAccountAccessSwitch();
+    let label = await accountAccessSwitch.getLabel();
+    expect(label).toEqual('Account enabled');
+
+    const saveButton = userProfileAdminPage.getSaveButton();
+    const cancelButton = userProfileAdminPage.getCancelButton();
+    await verifyButtonEnabled(cancelButton, false);
+    await verifyButtonEnabled(saveButton, false);
+
+    // Turn off Account Access toggle
+    await accountAccessSwitch.turnOff();
+    label = await accountAccessSwitch.getLabel();
+    expect(label).toEqual('Account disabled');
+
+    // Undo toggle change
+    await accountAccessSwitch.turnOn();
+    label = await accountAccessSwitch.getLabel();
+    expect(label).toEqual('Account enabled');
+
+    await verifyButtonEnabled(cancelButton, false);
+    await verifyButtonEnabled(saveButton, false);
+
+    // Change Initial Credits Limit to $400
+    const initialCreditLimit = userProfileAdminPage.getInitialCreditLimit();
+    const oldCreditLimit = await initialCreditLimit.getSelectedValue();
+    await initialCreditLimit.select('$400.00');
+
+    // Verify background color in Select has changed
+    const [styleElement] = await (await initialCreditLimit.asElement()).$x('./preceding-sibling::style');
+    const styleText = await getPropValue<string>(styleElement, 'innerText');
+    expect(styleText).toContain('background-color: #F8C954;');
+
+    await verifyButtonEnabled(cancelButton, true);
+    await verifyButtonEnabled(saveButton, true);
+
+    // Change Contact Email
+    const contactEmail = userProfileAdminPage.getContactEmail();
+    const oldContactEmail = await contactEmail.getValue();
+    // Input background color is white before change
+    let backgroundColor = await getStyleValue<string>(
+      adminTab,
+      await contactEmail.asElementHandle(),
+      'background-color'
+    );
+    expect(backgroundColor).toEqual('rgb(255, 255, 255)');
+
+    // Type in modified email
+    const invalidContactEmail = 'mod-' + oldContactEmail;
+    await contactEmail.type(invalidContactEmail);
+
+    // Input background color changed
+    backgroundColor = await getStyleValue<string>(adminTab, await contactEmail.asElementHandle(), 'background-color');
+    expect(backgroundColor).toEqual('rgb(248, 201, 84)');
+
+    // Find email error
```

Comment:
add something about how this is the erroor message we see when the Inst matches exact email addresses